pkg/reconcilermanager/controllers/reposync_controller_test.go

```diff
@@ -2867,11 +2888,17 @@ func validateClusterRoleBinding(want *rbacv1.ClusterRoleBinding, fakeClient *syn
 	return nil
 }
 
-// validateDeployments validates that important fields in the `wants` deployments match those same fields in the deployments found in the fakeClient
-func validateDeployments(wants map[core.ID]*appsv1.Deployment, fakeClient *syncerFake.Client) error {
+// validateDeployments validates that important fields in the `wants` deployments match those same fields in the current deployments found in the unstructured Map
+func validateDeployments(wants map[core.ID]*appsv1.Deployment, currents map[core.ID]*unstructured.Unstructured) error {
```

Comment:
Suggest passing the fake dynamic client as argument. From the caller's point of view, it doesn't need to know the underlying data of the fake client when calling the `validateDeployments` function.

Reply:
Done